src/Opulence/Databases/StatementException.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Opulence\Databases;
+
+use Exception;
+use PDOStatement;
+use Throwable;
+
+class StatementException extends Exception
+{
+    /** @var PDOStatement */
+    protected $statement;
+
+    /**
+     * StatementException constructor.
+     *
+     * @param PDOStatement   $statement
+     * @param string         $message
+     * @param int            $code
+     * @param Throwable|null $previous
+     */
+    public function __construct(PDOStatement $statement, $message = "", $code = 0, Throwable $previous = null)
+    {
+        $this->statement = $statement;
+
+        $message = $message ?: $statement->errorInfo()[2];
+
+        parent::__construct($message, $code, $previous);
+    }
+
+    /**
+     * @return PDOStatement
+     */
+    public function getStatement(): PDOStatement
+    {
+        return $this->statement;
+    }
+
+    /**
+     * @see https://www.php.net/manual/en/pdostatement.errorinfo.php
+     *
+     * @return array
+     */
+    public function getErrorInfo(): array
```

Comment:
Is there any reason to have these methods?  Should we just expose `getStatement()` and let users call `errorInfo()` and `errorCode()` on that?

Reply:
Not sure fully I agree, but works for me.